src/utils/components/DiskModal/components/DiskSizeInput/DiskSizeInput.tsx

```diff
@@ -0,0 +1,39 @@
+import React, { FC } from 'react';
+import { Controller, useFormContext } from 'react-hook-form';
+
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { FormGroup, TextInput } from '@patternfly/react-core';
+
+import CapacityInput from '../../../CapacityInput/CapacityInput';
+import { DiskFormState, SourceTypes } from '../../utils/types';
+import { DYNAMIC } from '../utils/constants';
+
+const DiskSizeInput: FC = () => {
+  const { t } = useKubevirtTranslation();
+  const { control, watch } = useFormContext<DiskFormState>();
+  const diskSource = watch('diskSource');
+
+  if (SourceTypes.PVC === diskSource || SourceTypes.OTHER === diskSource) {
+    return null;
+  }
+
+  if (SourceTypes.EPHEMERAL === diskSource) {
+    return (
+      <FormGroup fieldId="ephemeral-disk-size" label="Size">
+        <TextInput id="ephemeral-disk-size" isDisabled type="text" value={DYNAMIC} />
+      </FormGroup>
+    );
+  }
```

Comment:
i understand why u did it, can u find a different solution that isn't including nested component?

Reply:
do you mean the have a component mapper or switch? or if I misunderstood can you explain?

Reply:
export it :)